orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/api/RetrofitBakeryService.java

```diff
@@ -0,0 +1,66 @@
+package com.netflix.spinnaker.orca.bakery.api;
+
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.orca.bakery.api.manifests.BakeManifestRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import retrofit2.Call;
+import retrofit2.HttpException;
+import retrofit2.Response;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+/**
+ * An interface to the Bakery's REST API.
+ */
+@Component
+public final class RetrofitBakeryService implements BakeryService {
+
+  private final BakeryApi api;
+
+  public @Autowired RetrofitBakeryService(BakeryApi api) {
+    this.api = api;
+  }
+
+  @Override public Artifact bakeManifest(BakeManifestRequest bakeRequest) {
+    return retrofit(() -> api.bakeManifest(bakeRequest));
+  }
+
+  @Override
+  public BakeStatus createBake(String region, BakeRequest bake, String rebake) {
+    return retrofit(() -> api.createBake(region, bake, rebake));
+  }
+
+  @Override public BakeStatus lookupStatus(String region, String statusId) {
+    return retrofit(() -> api.lookupStatus(region, statusId));
+  }
+
+  @Override public Bake lookupBake(String region, String bakeId) {
+    return retrofit(() -> api.lookupBake(region, bakeId));
+  }
+
+  //
+  // Methods below this line are not supported by the Netflix Bakery, and are only available
+  // iff bakery.roscoApisEnabled is true.
+  //
+
+  @Override
+  public BaseImage getBaseImage(String cloudProvider, String imageId) {
+    return retrofit(() -> api.getBaseImage(cloudProvider, imageId));
+  }
+
+  private <T> T retrofit(Supplier<Call<T>> call) {
+    try {
+      Response<T> response = call.get().execute();
+      if (response.isSuccessful()) {
+        return response.body();
+      } else {
+        throw new HttpException(response);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
This probably warrants something a little cleaner ... brief look at docs show an `IOException` being somewhat equivalent to a `NETWORK` error.

Not sure if trying to replicate a similar exception is worthwhile ... https://gist.github.com/koesie10/bc6c62520401cc7c858f#file-retrofitexception-java

Reply:
I hadn't thought about the migration pain that comes with having to support both `RetrofitError` and `HttpException`. From a shared lib perspective, we've got... `RetrySupport` -- anything else? Too bad Java doesn't support implicit interfaces. :(

I'm not attached to the exact `RetrofitError` impl, but I do think wrapping `HttpException` / `IOException` in a common exception would be valuable.

Reply:
Yeah, I deliberately didn't sink too much time into this. We need to have a good exception strategy here. I really just wanted to see what you guys thought of this approach of wrapping a Retrofit interface with a (very similar) service interface.

Reply:
Thus far `RetrySupport` doesn't really factor into anything retrofit-related. 

At least as far as `orca` goes, it should be reasonably easy to track down everywhere that RetrofitError is currently being used.

In 4 years of hindsight, it would have been nice to have wrapped RetrofitError with something of our own.